### PR TITLE
Support choosing partition by schema or table

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Support choosing partition by schema or table if users explicitly say so
otherwise use fixed partition 0.

Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

### What problem does this PR solve? 
Make drainer possible to utilize multiple kafka partitions when the global ordering of binlog events isn't  
required.

### What is changed and how it works?
Code changes
Support choosing kafka partition by schema or table.

 - Has exported function/method change
no
 - Has exported variable/fields change
no
 - Has interface methods change
no
 - Has persistent data change
no

Related changes
 - Need to cherry-pick to the release branch
no
 - Need to update the documentation
yes
 - Need to update the `tidb-ansible` repository
yes
 - Need to be included in the release note
no

### CODE REVIEW

**Comment by sunxiaoguang** [comment]
Looks like integration test failed in a strange way.

File: drainer/sync/kafka.go [hunk 1]
```diff
@@ -32,6 +32,18 @@ var maxWaitTimeToSendMSG = time.Second * 30
 
 var _ Syncer = &KafkaSyncer{}
 
+// PartitionMode Kafka partition mode
+type PartitionMode byte
+
+const (
+	// PartitionFixed choose fixed partition 0
+	PartitionFixed = iota + 1
```
Comment (kennytm) [inline drainer/sync/kafka.go]:
```suggestion
	PartitionFixed PartitionMode = iota + 1
```

File: drainer/sync/kafka.go [hunk 2]
```diff
@@ -112,12 +142,60 @@ func (p *KafkaSyncer) Sync(item *Item) error {
...
+		}
+	}
+	return nil
+}
+
+func (p *KafkaSyncer) syncDMLPartitionByTable(binlog *obinlog.Binlog, item *Item) error {
+	tables := binlog.DmlData.Tables
+	if len(tables) == 1 {
+		table := tables[0]
+		return errors.Trace(p.saveBinlog(binlog, item, sarama.StringEncoder(*table.SchemaName+"."+*table.TableName)))
```
Comment (kennytm) [inline drainer/sync/kafka.go]: This specialization doesn't seem useful. The outcome of `len(tables) == 1` and `len(tables) != 1` are exactly the same.

Comment (sunxiaoguang) [reply drainer/sync/kafka.go]: This can save one allocation. But it's small though and probably can be ignored.

File: drainer/sync/kafka.go (hunk 1 above)
Comment (sunxiaoguang) [reply drainer/sync/kafka.go]: Fixed

**Pull request closed without merging**